### PR TITLE
Update NCO invocations

### LINKED ISSUE #254: Potential update for NCO invocation
Potential update for NCO invocation -- see if `zppy` needs similar changes to https://github.com/E3SM-Project/e3sm_diags/pull/593

### PR DESCRIPTION
**Pull request by forsyth2**
Update NCO invocations. Resolves #254

### CODE REVIEW

File: zppy/templates/ts.bash [hunk 1]
```diff
@@ -65,6 +65,7 @@ cat input.txt | ncclimo \
 -c {{ case }} \
 -v {{ vars }} \
 --mem_mb=0 \
+--split \
```
Comment (forsyth2) [inline zppy/templates/ts.bash:68]: Adding `--split`, as in https://github.com/E3SM-Project/e3sm_diags/pull/593.

Comment (forsyth2) [reply zppy/templates/ts.bash:68]: @chengzhuzhang @czender Even with this change, I'm getting the following. I wasn't running into any issues with the previous Unified's version of NCO.
```
$ tail -n 11 ts_land_monthly_1850-1851-0002.o190591
ncremap: ERROR Failed to horizontally regrid. cmd_rgr[0] failed. Debug this:
 ncks -O -t 1 --no_tmp_fl --hdr_pad=10000 --no_tmp_fl --hdr_pad=10000 --gaa remap_script=ncremap --gaa remap_hostname=chr-0493 --gaa remap_version=5.0.7    --rgr lat_nm_out=lat --rgr lon_nm_out=lon --rgr no_stagger --rgr sgs_frc_nm=landfrac#sgs_msk_nm=landmask --map_fl="/home/ac.zender/data/maps/map_ne30pg2_to_cmip6_180x360_aave.20200201.nc" "trash/FSH_185001_185112.nc" "output/FSH_185001_185112.nc"
ncclimo: ERROR Failed to regrid. cmd_rgr[0] failed. Debug this:
 ncremap -u .pid602674.split.0.tmp --nco_opt="--no_tmp_fl --hdr_pad=10000" -t 1  --no_stagger -P sgs  --map=/home/ac.zender/data/maps/map_ne30pg2_to_cmip6_180x360_aave.20200201.nc trash/FSH_185001_185112.nc output/FSH_185001_185112.nc
/lcrc/soft/climate/e3sm-unified/spack/spack_for_mache_1.4.1/var/spack/environments/e3sm_unified_1_7_0rc1_chrysalis_gnu_openmpi/.spack-env/view/bin/ncclimo: line 2836: kill: (602753) - No such process
/lcrc/soft/climate/e3sm-unified/spack/spack_for_mache_1.4.1/var/spack/environments/e3sm_unified_1_7_0rc1_chrysalis_gnu_openmpi/.spack-env/view/bin/ncclimo: line 2836: kill: (602754) - No such process
+ '[' 1 '!=' 0 ']'
+ cd /lcrc/group/e3sm/ac.forsyth2/zppy_test_complete_run_output/v2.LR.historical_0201/post/scripts
+ echo 'ERROR (1)'
+ exit 1
ncks: ERROR nco_rgr_wgt() (aka "the regridder") reports unable to find sgs_frc_nm = landfrac in current input file, and unable to identify filename (ending with slash '/' or backslash '\', depending on the operating system) portion of that string to serve as local external file for sgs_frc input, exiting
```

Comment (czender) [reply zppy/templates/ts.bash:68]: My interpretation of this error comes from the error message itself. It appears that the input file `trash/FSH_185001_185112.nc` does not contain the `landfrac` variable that `ncremap` needs to regrid ELM files correctly. There are two routes you can take: either make sure the workflow retains `landfrac` in each input file until `ncremap` is invoked, or, alternatively use the `--sgs_frc=/path_to_file_with_landfrac/filename/landfrac` to point `ncremap` to a valid `landfrac` in a second file. Does this explain the problem you encountered? If not, please provide full commands and leave the inputs where I can find them.

File: tests/integration/test_complete_run.cfg [hunk 2]
```diff
@@ -85,7 +87,10 @@ years = "1850:1854:2", "1850:1854:4",
   [[ atm_monthly_180x360_aave ]]
   climo_diurnal_frequency = "diurnal_8xdaily"
   climo_diurnal_subsection = "atm_monthly_diurnal_8xdaily_180x360_aave"
+  environment_commands = "source /home/$USER/miniconda3/etc/profile.d/conda.sh; conda activate e3sm_diags_dev_20220614"
+  partition = compute
   sets = "lat_lon","zonal_mean_xy","zonal_mean_2d","polar","cosp_histogram","meridional_mean_2d","enso_diags","qbo","diurnal_cycle","annual_cycle_zonal_mean","streamflow", "zonal_mean_2d_stratosphere",
+  walltime = "01:00:00"
```
Comment (forsyth2) [inline tests/integration/test_complete_run.cfg:93]: E3SM Diags now seems to take ~50 minutes rather than under 30 minutes.

**Review by forsyth2 — COMMENTED** [review]
@chengzhuzhang I seem to have this working now. 

It fails the bash-generation test, as expected (since `--split` was added).

However, I'm also seeing many changed files for the complete-run test (https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.forsyth2/image_check_failures/20220622_image_check_failures_complete_run/). Is that expected given updates in E3SM Diags and Unified overall? I'm looking through the image check failures to see how significant they are.

File: tests/integration/test_complete_run.cfg (hunk 2 above)
Comment (chengzhuzhang) [reply tests/integration/test_complete_run.cfg:93]: Longer run time is due to the additional ERA sets.

**Comment by chengzhuzhang** [comment]
@forsyth2 Thanks for providing the images, the E3SM_diags figures seems as expected. I think the images are very helpful. I notice there are some changes in MPAS-A plots, aspect ratio of the figures is changed, maybe @xylar can confirm if this is expected. Also the starting date for some MPAS-A time series plots changes, which I suppose was caused by zppy configuration.

**Comment by forsyth2** [comment]
> aspect ratio of the figures is changed, maybe @xylar can confirm if this is expected

Do you have an example figure?

> the starting date for some MPAS-A time series plots changes

Do you have an example figure for this as well?

There were a lot of diffs in https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.forsyth2/image_check_failures/20220622_image_check_failures_complete_run/, so I could only check a handful. 

One thing I noticed in a a decent number of plots: https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.forsyth2/image_check_failures/20220622_image_check_failures_complete_run/fixed_salinArgo_v2_actual.png: the plots in the image are cut off. Also true in expected version. The actual image displayed is fine though (e.g., https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.forsyth2/zppy_test_complete_run_www/v2.LR.historical_0201/mpas_analysis/ts_1850-1854_climo_1850-1854/ocean/index.html#global_sss&gid=10&pid=24 but note that link is for a more recent run).

**Comment by xylar** [comment]
> I notice there are some changes in MPAS-A plots, aspect ratio of the figures is changed, maybe @xylar can confirm if this is expected. 

To be honest, you all pay a lot more attention to this than I am able to.  I simply accept that the figures change in subtle ways as matplotlib, cartopy, etc. get updated.  It's a full-time job trying to keep them from changing.

**Comment by forsyth2** [comment]
@xylar Ok, just wanted to make sure there were no egregious changes. Thanks!

**Comment by xylar** [comment]
@forsyth2, the image you linked to looks like one of the thumbnails we use for quick links.  They are cropped on purpose.  Here is the directory of thumbnails in the analysis you pointed to: https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.forsyth2/zppy_test_complete_run_www/v2.LR.historical_0201/mpas_analysis/ts_1850-1854_climo_1850-1854/ocean/thumbnails/

**Comment by chengzhuzhang** [comment]
@forsyth2 here is an example of starting time change:
https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.forsyth2/image_check_failures/20220622_image_check_failures_complete_run/TAnomalyZ_global_v2_actual.png
versus
https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.forsyth2/image_check_failures/20220622_image_check_failures_complete_run/TAnomalyZ_global_v2_expected.png

It looks like the first figure lost 3 month at the begining

**Comment by chengzhuzhang** [comment]
@forsyth2 I'm not exactly sure about what causes the difference. Hopefully it is not coming from the changes made in this PR.

**Comment by forsyth2** [comment]
@chengzhuzhang I ran `main` but with a `cfg` that only ran MPAS-Analysis using `environment_commands = "source /lcrc/soft/climate/e3sm-unified/test_e3sm_unified_1.7.0rc1_chrysalis.sh"`. The dates still changed. So, I'm confident the date change is related to MPAS-Analysis and not `zppy` directly.

**Review by chengzhuzhang — APPROVED** [review]

**Comment by chengzhuzhang** [comment]
Hi @xylar Thank you for confirming on this, so that we can understand the cause! I don't think it causes trouble or needs a fix.

**Pull request merged**